/* Platform Device Control Interface - hatch2sr */

//First: 0x10000120
//Last:  0x10000129

syntax = "proto3";

import "icon/metadata/metadata.proto";

package pdci.hatch;

enum HatchStatus {
  Undefined = 0;
  Open = 1;
  Closed = 2;
  ChangingPosition = 3;
  Faulty = 4;
}

message GetStatusReq
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000120;
  string device_id = 1;
}

message GetStatusCfm
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000121;
  HatchStatus status = 1;
}

message OpenHatchReq
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000122;
  string device_id = 1;
}

message OpenHatchCfm
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000123;
}

message CloseHatchReq
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000124;
  string device_id = 1;
}

message CloseHatchCfm
{
  option (icon.metadata.MESSAGE_NUMBER) = 0x10000125;
}